13.3.5: pytest is failing because it uses no longer maintained `random2` module

* Faker version: 13.3.5
* OS: Linux x86/64

### Brief summary of the issue
Looks like pytest is failing because test suite still is using outdated `random2` module which was never updated for python 3.x).

### Steps to reproduce
I'm trying to package your module as an rpm package. So I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

### Expected behavior
pytest should not fail.

### Actual behavior
Here is pytest output:
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-faker-13.3.5-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-faker-13.3.5-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -q
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.13, pytest-7.1.1, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/faker-13.3.5, configfile: setup.cfg
plugins: Faker-13.3.5
collected 1491 items / 1 error

================================================================================== ERRORS ==================================================================================
_______________________________________________________________ ERROR collecting tests/providers/test_ssn.py _______________________________________________________________
ImportError while importing test module '/home/tkloczko/rpmbuild/BUILD/faker-13.3.5/tests/providers/test_ssn.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib64/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/providers/test_ssn.py:11: in <module>
    import random2
E   ModuleNotFoundError: No module named 'random2'
============================================================================= warnings summary =============================================================================
../../BUILDROOT/python-faker-13.3.5-2.fc35.x86_64/usr/lib/python3.8/site-packages/faker/providers/person/fr_QC/__init__.py:10
  /home/tkloczko/rpmbuild/BUILDROOT/python-faker-13.3.5-2.fc35.x86_64/usr/lib/python3.8/site-packages/faker/providers/person/fr_QC/__init__.py:10: UserWarning: fr_QC locale is deprecated. Please use fr_CA.
    warnings.warn("fr_QC locale is deprecated. Please use fr_CA.")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= short test summary info ==========================================================================
ERROR tests/providers/test_ssn.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================================= 1 warning, 1 error in 10.95s =======================================================================
```

--- a/tests/providers/test_ssn.py
+++ b/tests/providers/test_ssn.py
@@ -9,7 +9,6 @@ from unittest import mock
 
 import freezegun
 import pytest
-import random2
 
 from validators.i18n.es import es_cif as is_cif
 from validators.i18n.es import es_nie as is_nie
@@ -300,7 +299,6 @@ class TestEnUS(unittest.TestCase):
             assert area != "666"
 
     def test_invalid_ssn(self):
-        self.fake.random = random2.Random()
         # Magic Numbers below generate '666-92-7944', '000-54-2963', '956-GG-9478', '436-00-1386',
         # and 134-76-0000 respectively. The "group" (GG) returned for '956-GG-9478 will be a random
         # number, and that random number is not in the "itin_group_numbers" List. The random GG occurs
@@ -362,7 +360,7 @@ class TestEnUS(unittest.TestCase):
             99,
         ]
 
-        self.fake.seed_instance(1143)
+        self.fake.seed_instance(2432)
         ssn = self.fake.ssn(taxpayer_identification_number_type="INVALID_SSN")
 
         assert len(ssn) == 11
@@ -380,7 +378,7 @@ class TestEnUS(unittest.TestCase):
 
         assert 900 <= int(area) <= 999 and int(group) not in itin_group_numbers
 
-        self.fake.seed_instance(9)
+        self.fake.seed_instance(0)
         ssn = self.fake.ssn(taxpayer_identification_number_type="INVALID_SSN")
         [area, group, serial] = ssn.split("-")
 
@@ -694,6 +692,7 @@ class TestEtEE(unittest.TestCase):
 
     def setUp(self):
         self.fake = Faker("et_EE")
+
         Faker.seed(0)
 
     def test_ssn_checksum(self):
@@ -704,14 +703,12 @@ class TestEtEE(unittest.TestCase):
 
     @freezegun.freeze_time("2019-03-11")
     def test_ssn(self):
-        self.fake.random = random2.Random()
-
-        self.fake.seed_instance(0)
+        self.fake.seed_instance(1)
         value = self.fake.ssn()
         assert re.search(r"^\d{11}$", value)
         assert not value.endswith("0")
 
-        self.fake.seed_instance(5)
+        self.fake.seed_instance(0)
         value = self.fake.ssn()
 
         assert re.search(r"^\d{11}$", value)
@@ -719,8 +716,6 @@ class TestEtEE(unittest.TestCase):
 
     @freezegun.freeze_time("2002-01-01")
     def test_ssn_2000(self):
-        self.fake.random = random2.Random()
-
         self.fake.seed_instance(0)
         value = self.fake.ssn(min_age=0, max_age=1)
         assert re.search(r"^\d{11}$", value)
@@ -728,8 +723,6 @@ class TestEtEE(unittest.TestCase):
 
     @freezegun.freeze_time("2101-01-01")
     def test_ssn_2100(self):
-        self.fake.random = random2.Random()
-
         self.fake.seed_instance(0)
         value = self.fake.ssn(min_age=0, max_age=1)
         assert re.search(r"^\d{11}$", value)

--- a/tox.ini
+++ b/tox.ini
@@ -7,7 +7,6 @@ deps =
     coverage>=5.2
     freezegun<0.4
     pytest>=6.0.1
-    random2>=1.0.1
     ukpostcodeparser>=1.1.1
     validators>=0.13.0
     sphinx>=2.4,<3.0

